File: answers/other/in sap/First Missing Positive.py
class Solution(object):
    def firstMissingPositive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        i = 0
        while i < len(nums):
            if nums[i] > 0 and nums[i] < len(nums) and nums[i] != i + 1 and nums[i] != nums[nums[i] - 1]:
                temp = nums[nums[i] - 1]
                nums[nums[i] - 1] = nums[i]
                nums[i] = temp
            else:
                i += 1
        for i in range(len(nums)):
            if nums[i] != i + 1:
                return i + 1
        return len(nums) + 1

File: answers/other/in sap/First Missing Positive.cpp
//@result 156 / 156 test cases passed. Status: Accepted Runtime: 4 ms Submitted: 0 minutes ago You are here! Your runtime beats 5.47% of cpp submissions.

class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        for (size_t i = 0; i < nums.size(); ++ i) {
            if (nums[i] > 0 && nums[i] <= nums.size() && nums[nums[i] - 1] != nums[i]) {
                int temp = nums[nums[i] - 1];
                nums[nums[i] - 1] = nums[i];
                nums[i] = temp;
                -- i;
            }
        }
        for (size_t i = 0; i < nums.size(); ++ i) {
            if (nums[i] - 1 != i) {
                return i + 1;
            }
        }
        return nums.size() + 1;
    }
};